src/codemagic/tools/app_store_connect.py

```diff
@@ -492,25 +492,53 @@ def list_devices(
         "register-device",
         BundleIdArgument.PLATFORM,
         DeviceArgument.DEVICE_NAME,
-        DeviceArgument.DEVICE_UDID,
+        DeviceArgument.DEVICE_UDIDS,
+        DeviceArgument.IGNORE_REGISTRATION_ERRORS,
     )
     def register_device(
         self,
         platform: BundleIdPlatform,
         device_name: str,
-        device_udid: str,
+        device_udids_argument: Union[str, Types.DeviceUdidsArgument, Sequence[Union[str, Types.DeviceUdidsArgument]]],
```

Comment:
Given that `DeviceArgument.DEVICE_UDIDS` is not required, then the action implementation should accept `None` values too and raise an error if the value is missing. Currently execution fails with an unexpected exception if no UDIDs are specified.